pkg/internal/common/logging/metadata/callerInfo.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package metadata
+
+import (
+	"github.com/hyperledger/aries-framework-go/pkg/common/logging/api"
+)
+
+type callerInfoKey struct {
+	module string
+	level  api.Level
+}
+
+//CallerInfo maintains module-level based information to show or hide caller info
+type CallerInfo struct {
+	showcaller map[callerInfoKey]bool
+}
+
+//ShowCallerInfo enables caller info for given module and level
+func (l *CallerInfo) ShowCallerInfo(module string, level api.Level) {
+	if l.showcaller == nil {
```

Comment:
This is done in three places - maybe create a func or initialize in a NewCallerInfo() func?